### ISSUE TITLE
[All][Templating] Custom token in adaptive card designer

### ISSUE THREAD

**Issue by viperfx** [microsoft/AdaptiveCards#3624]
**What platform is your feature request related to?**

All

**Is your feature request related to a problem? Please describe.**
Currently, the token is double single braces `{VAR}`. This can be different when porting the templating to another language and contributing an SDK. Especially if the language has an existing templating library. 

**Describe the solution you'd like**
Allow for a custom token such as double curly braces or anything the user wishes to set. 

This should be editable in the designer, to make prototyping easier.

Other templating languages have a set token, so the implementor for a new SDK may not have a choice and this would allow more people to adopt the new templating feature.

**Comment by matthidinger** [comment]
Hi @viperfx, thanks for starting this conversation! This is an interesting topic that I'd love to get your thoughts on.

Is there a technical limitation in your Python environment that is preventing the current syntax of `{expression}`?

While I can totally appreciate your request for making the binding syntax configurable at runtime and within the designer, that does introduce two problems:

- It introduces a layer of complexity and incompatibility between SDKs. 
- In some ways this works against our goals for the standardized template language when client-side templating becomes a reality. Our hope in the future is that a template can be downloaded directly on a client (like Teams), and the data payload can also be downloaded separately. This would enable a scenario where the card content could be updated rapidly by sending updated data to the client, reusing the same template. I can't think of a way to make this work without having a standard binding expression that works across all SDKs and Host Apps.

Based on this, I'd love to understand if we can make it work with Python, even if that means re-evaluating the escape syntax from a single curly brace to something else.

@DeeJayTC FYI in case you had thoughts too

**Comment by viperfx** [comment]
Hey Matt, 

Thanks for the detailed reply. 

My main reason for a custom token was because we use Django internally. It comes with a template language but does not have the option for a custom token. 

If it is introducing a lot of complexity, then I think there are other options out there. Such as Jinja2 (https://jinja.palletsprojects.com/en/2.10.x/api/#jinja2.Environment) allowing to customise the tokens before parsing. 

However due to this JSON-to-JSON representation, on the second review of the existing JS Template engine, I think python Templating engines such as Jinja, or Django, would not fit with the overall goal and compatibility you are going for with this templating feature. These template languages will require the payload to be modified (https://jinja.palletsprojects.com/en/2.10.x/templates/) with control and flow statements. Hence the payload can no longer be pasted into the designer. 

In conclusion, I think the only way forward is a port of the templating engine to python

**Comment by RebeccaAnne** [comment]
This is a good suggestion, but not something that's on our roadmap at this time. We may reconsider it in the future, or if you would like to make a contribution please see our contribution guidelines [here](https://github.com/microsoft/AdaptiveCards/blob/main/.github/CONTRIBUTING.md).
